### DISCUSSION TITLE
v1.12.2 (Beta)

### DISCUSSION THREAD

**Discussion by deividAlfa** [deividAlfa/stm32_soldering_iron_controller#90] [Announcements]
- I hope it works now... For some reason my station works fine, so it's pretty hard to find them out.

**v1.12.1 changelog:**
- Hopefully all issues from v1.12.0 related to Profile/Temperature/Tip not being saved are fixed now.

**v1.12.0 changelog:**
- **Massive** rework of the entire firmware to extend the future of 10KB devices, optimizing the ram usage by a lot.
  Few days ago they had less than 80 bytes of RAM left, so these deviced were seriously limited.
  Now they have 1.5KB free, opening a lot of possibilities.
  No longer store the entire profile in RAM, only a small bunch of common settings and the current tip.
  Now all the settings (System, Profile, Addons, Tip) use only about 260 Bytes of RAM.

- Recent Hardfault error from 1.11.8/9/10 is finally gone ?

- Some improvements to the tip editor:
  When editing a tip, modiy the existing name and later click on Copy, this will transfer the name to the new tip, creating new tips much easier.
   *T12-BC3 -> T12-D12 -> Copy -> New tip will be called T12-D12*.
  If the tip name was left unmodified, the new tip name will be blank.

  There's an inmense amount of changes going under hood, not really making any difference at the user level.
Due the nature of the changes, all the setting structures were modified.

  Thus, **this version will wipe all the existing data!** (Hopefully it will be the last time it happens in a long time).

     Please report any bugs found!

**Comment by sst311212** [comment]
Tip's cal data will not save to flash without save_Tip attribute, and it also need adding this before saveSettings.
```
systemSettings.tipUpdateMode = mode_SaveTip;
systemSettings.tipUpdateIndex = systemSettings.currentTip;
```
https://github.com/deividAlfa/stm32_soldering_iron_controller/blob/18b4558dec27c2247817248ff06922f12b6b3d8a/Drivers/graphics/gui/screens/calibration_screen.c#L335-L342


This will select current tip instead of wanted tip in combo menu.
https://github.com/deividAlfa/stm32_soldering_iron_controller/blob/18b4558dec27c2247817248ff06922f12b6b3d8a/Drivers/graphics/gui/screens/tip_settings_screen.c#L223-L226

**Comment by deividAlfa** [comment]
Both fixed now

**Comment by deividAlfa** [comment]
Locked as this is an old release, bugs where fixed.
